data/scenarios/Challenges/wave.yaml

```diff
@@ -0,0 +1,74 @@
+      not: got_caught
+    condition: |
+      as base {has "bitcoin"};
+  - id: got_caught
+    teaser: Got caught
+    optional: true
+    hidden: true
+    goal:
+      - |
+        The robots caught you!
+    condition: |
+      as base {x <- meet; return $ case x (\_. false) (\_. true)};
+robots:
+  - name: base
+    dir: [0, 1]
+    devices:
+      - branch predictor
+      - comparator
+      - dictionary
+      - grabber
+      - hourglass
+      - lambda
+      - logger
+      - net
+      - scanner
+      - strange loop
+      - treads
+  - name: wavebot
+    system: true
+    dir: [0, 1]
+    display:
+      invisible: false
+      attr: 'plant'
+    program: |
+      run "scenarios/Challenges/_wave/wavebot.sw"
+solution: |
+   run "scenarios/Challenges/_wave/solution.sw"
+entities: []
+known: [wavy water, water, bitcoin]
+world:
+  dsl: |
+    overlay
+    [ {dirt, water}
+    , if x % 7 + y % 5 == 0 then {dirt, wavy water} else {blank}
+    ]
+  upperleft: [-3, 6]
+  offset: false
+  palette:
+    'B': [grass, erase, base]
+    'w': [grass, erase, wavebot]
+    'x': [dirt, water]
+    'z': [grass, bitcoin]
+    '.': [grass, erase]
+  map: |
+    xxxxxx..................................................................................................................................................................................................................................................xxxxxx
+    xxx........................................................................................................................................................................................................................................................xxx
+    x............................................................................................................................................................................................................................................................x
+    B............................................................................................................................................................................................................................................................z
+    x............................................................................................................................................................................................................................................................x
+    xxx........................................................................................................................................................................................................................................................xxx
+    xxxxxxwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwxxxxxx
+    
```

Comment:
Maybe this could be a bit shorter? Though it does work well as a performance test. 😅